projects/core/src/auth/user-auth/services/auth-state-persistence.service.ts

```diff
@@ -90,10 +90,14 @@ export class AuthStatePersistenceService implements OnDestroy {
       }
       if (state.userId) {
         this.userIdService.setUserId(state.userId);
+      } else {
+        this.userIdService.clearUserId();
       }
       if (state.redirectUrl) {
         this.authRedirectStorageService.setRedirectUrl(state.redirectUrl);
       }
+    } else {
+      this.userIdService.clearUserId();
```

Comment:
Was thinking about making it a bit more readable, but the best I could come up with is:
```
    if (state) {
      if (state.token) {
        this.authStorageService.setToken(state.token);
      }
      if (state.redirectUrl) {
        this.authRedirectStorageService.setRedirectUrl(state.redirectUrl);
      }
    }
    if (state?.userId) {
      this.userIdService.setUserId(state.userId);
    } else {
      this.userIdService.clearUserId();
    }
```
So leaving it up to you